sdk/auth/managed_identity_authorizer.go

```diff
@@ -156,7 +157,7 @@ func (c *managedIdentityConfig) TokenSource(_ context.Context) (Authorizer, erro
 	})
 }
 
-func azureMetadata(ctx context.Context, url string) (body []byte, err error) {
+func azureMetadata(ctx context.Context, url string) (body []byte, statusCode int, err error) {
 	ctx2, cancel := context.WithDeadline(ctx, time.Now().Add(time.Second*30))
 	defer cancel()
```

Comment:
there's a legitimate issue here however, where the context deadline may be exceeded before the request has completed, should the IDMS service become unavailable (where it can have a 70 second delay) - so we should instead push this up:

```suggestion
	if _, ok := ctx.Deadline(); !ok {
		return nil, fmt.Errorf("internal-error: context had no deadline")
	}
```

which'll require calls to the token function to have a context with a deadline attached, which is the expected behaviour here here